### PR TITLE
Feature/se 1016.upsert.simple.order.tutorial

### PR DESCRIPTION
**Pull request by OmarFinbourne**
Merge request to create a test to upsert a simple order

### CODE REVIEW

File: sdk/tests/tutorials/ibor/test_orders.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                                     instrument_identifiers=instrument_identifiers,
+                                     quantity=quantity, side='buy',
+                                     order_book_id=order_book_id,
+                                     portfolio_id=portfolio_id, id=order_id)
+
+        order_req = OrderSetRequest(order_requests=[order_request])
+
+        upsert_result = self.orders_api.upsert_orders(order_set_request=order_req)
+
+        assert len(upsert_result.values) == 1
```
Comment (rizwansaeed) [inline sdk/tests/tutorials/ibor/test_orders.py:61]: use the `unitest` `assertEqual` and the suite of `assert...` functions

File: sdk/tests/tutorials/ibor/test_orders.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        assert upsert_result.values[0].to_dict()['id']['code'] == order
+        assert upsert_result.values[0].to_dict()['lusid_instrument_id'] == self.instrument_ids[0]
+        assert upsert_result.values[0].to_dict()['quantity'] == 100
+        assert upsert_result.values[0].to_dict()['properties'][f"Order/{test_scope}/TIF"]['key'] == f"Order/{test_scope}/TIF"
+
```
Comment (rizwansaeed) [inline sdk/tests/tutorials/ibor/test_orders.py:66]: delete the order at the end of the test

File: sdk/tests/tutorials/ibor/test_orders.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        # create a configured API client
+        api_client = TestDataUtilities.api_client()
+        cls.orders_api = lusid.OrdersApi(api_client)
+        cls.instruments_api = lusid.InstrumentsApi(api_client)
+        instrument_loader = InstrumentLoader(cls.instruments_api)
+        cls.instrument_ids = instrument_loader.load_instruments()
+
+    def get_guid(self):
+        # creates random alphanumeric code
+        return str(uuid.uuid4())[:12]
```
Comment (rizwansaeed) [inline sdk/tests/tutorials/ibor/test_orders.py:22]: don't truncate the guids, it makes them a lot less unique!

Comment (rizwansaeed) [reply sdk/tests/tutorials/ibor/test_orders.py:22]: also this is trivial enough it doesn't need to be a function (once you remove the truncation)

File: sdk/tests/tutorials/ibor/test_orders.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    @lusid_feature("F4")
+    def test_upsert_simple_order(self):
+        """Test makes a request for a single order."""
```
Comment (rizwansaeed) [inline sdk/tests/tutorials/ibor/test_orders.py:26]: whilst this is technically a test, in reality its sample code, as such you can remove "Test" from your docstring

File: sdk/tests/tutorials/ibor/test_orders.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        test_scope = "Orders-SimpleUpsert-TestScope"
+        # Create unique order id
+        order = self.get_guid()
+        # Create ResourceId for order
+
+        # Construct order arguments
+        instrument_identifiers = {TestDataUtilities.lusid_luid_identifier: self.instrument_ids[0]}
+        order_id = models.ResourceId(test_scope, order)
+        portfolio_id = ResourceId(test_scope, "OrdersTestPortfolio")
+        # The key:This takes the format {domain}/{scope}/{code}, value: Lusid property value'
```
Comment (rizwansaeed) [inline sdk/tests/tutorials/ibor/test_orders.py:37]: I'm not sure this comment is useful (for a reader), its duplicating the actual api docs

File: sdk/tests/tutorials/ibor/test_orders.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+    def get_guid(self):
+        # creates random alphanumeric code
+        return str(uuid.uuid4())[:12]
+
+    @lusid_feature("F4")
+    def test_upsert_simple_order(self):
+        """Test makes a request for a single order."""
+
+        test_scope = "Orders-SimpleUpsert-TestScope"
```
Comment (rizwansaeed) [inline sdk/tests/tutorials/ibor/test_orders.py:28]: rename this to something like `orders_scope`, remember these are examples so we should try and use useful names i.e. avoid names like test, demo etc.

File: sdk/tests/tutorials/ibor/test_orders.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
+import unittest
+import uuid
+import lusid
+import lusid.models as models
+from lusid import *
```
Comment (rizwansaeed) [inline sdk/tests/tutorials/ibor/test_orders.py:5]: fix the sonar break, only import what you need

Comment (OmarFinbourne) [reply sdk/tests/tutorials/ibor/test_orders.py]: Thanks, I cleaned this up

File: sdk/tests/tutorials/ibor/test_orders.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+                                     quantity=quantity, side='buy',
+                                     order_book_id=order_book_id,
+                                     portfolio_id=portfolio_id, id=order_id)
+
+        order_req = OrderSetRequest(order_requests=[order_request])
+
+        upsert_result = self.orders_api.upsert_orders(order_set_request=order_req)
+
+        self.assertEqual(len(upsert_result.values),1)
+        self.assertEqual(upsert_result.values[0].to_dict()['id']['code'], order)
```
Comment (kkeenan-finbourne) [inline sdk/tests/tutorials/ibor/test_orders.py:61]: As your reusing `upsert_result.values[0].to_dict()`  in your assertions you should just create a variable for the result. Save you converting it to a to_dict everytime.

Also worth checking do you actually need to convert it to a dict via to_dict()?

File: sdk/tests/tutorials/ibor/test_orders.py [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+        order_book_id = ResourceId(orders_scope, "OrdersTestBook")
+        quantity = 100
+        # Construct request
+        order_request = OrderRequest(properties=properties,
+                                     instrument_identifiers=instrument_identifiers,
+                                     quantity=quantity, side='buy',
+                                     order_book_id=order_book_id,
+                                     portfolio_id=portfolio_id, id=order_id)
+
+        order_req = OrderSetRequest(order_requests=[order_request])
```
Comment (kkeenan-finbourne) [inline sdk/tests/tutorials/ibor/test_orders.py:56]: Perhaps look at renaming `order_req` to `order_set_request`. Prevents the confusion between the OrderSetRequest with the actual underlying OrderRequests

File: sdk/tests/tutorials/ibor/test_orders.py [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+        cls.instrument_ids = instrument_loader.load_instruments()
+
+ #   @lusid_feature("F4")
+    def test_upsert_simple_order(self):
+        """Makes a request for a single order."""
+
+        orders_scope = "Orders-SimpleUpsert-TestScope"
+        # Create unique order id
+        order = str(uuid.uuid4())
+        # Create ResourceId for order
```
Comment (kkeenan-finbourne) [inline sdk/tests/tutorials/ibor/test_orders.py:31]: Should this comment be moved to before line 35?

File: sdk/tests/tutorials/ibor/test_orders.py [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+
+        # Construct order arguments
+        instrument_identifiers = {TestDataUtilities.lusid_luid_identifier: self.instrument_ids[0]}
+        order_id = models.ResourceId(orders_scope, order)
```
Comment (kkeenan-finbourne) [inline sdk/tests/tutorials/ibor/test_orders.py:35]: Rename `order_id` to `order_resource_id` and then you can rename `order` on line 30 to `order_id`. Calling it order is a bit confusing as you'd assume it's a full order object and not just the id

File: sdk/tests/tutorials/ibor/test_orders.py [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+from lusid import OrderSetRequest
+from lusid import PerpetualProperty
+from lusid import PropertyValue
+from lusid import ResourceId
+from utilities import InstrumentLoader
+from utilities import TestDataUtilities
+
+
+class Orders(unittest.TestCase):
+    @classmethod
```
Comment (kkeenan-finbourne) [inline sdk/tests/tutorials/ibor/test_orders.py:15]: Just cosmetic but add a line break here to keep consistent with the other tests

File: sdk/tests/tutorials/ibor/test_orders.py [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+
+        order_req = OrderSetRequest(order_requests=[order_request])
+
+        upsert_result = self.orders_api.upsert_orders(order_set_request=order_req)
+
+        self.assertEqual(len(upsert_result.values),1)
+        self.assertEqual(upsert_result.values[0].to_dict()['id']['code'], order)
+        self.assertEqual(upsert_result.values[0].to_dict()['lusid_instrument_id'], self.instrument_ids[0])
+        self.assertEqual(upsert_result.values[0].to_dict()['quantity'], 100)
+        self.assertEqual(upsert_result.values[0].to_dict()['properties'][f"Order/{orders_scope}/TIF"]['key'] , f"Order/{orders_scope}/TIF")
```
Comment (kkeenan-finbourne) [inline sdk/tests/tutorials/ibor/test_orders.py:64]: Really like the additional assertion here to ensure properties have been set

**Review by kkeenan-finbourne — APPROVED** [review]

File: sdk/tests/tutorials/ibor/test_orders.py [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+    @classmethod
+    def setUpClass(cls):
+        # create a configured API client
+        api_client = TestDataUtilities.api_client()
+        cls.orders_api = lusid.OrdersApi(api_client)
+        cls.instruments_api = lusid.InstrumentsApi(api_client)
+        instrument_loader = InstrumentLoader(cls.instruments_api)
+        cls.instrument_ids = instrument_loader.load_instruments()
+
+   @lusid_feature("F4")
```
Comment (rizwansaeed) [inline sdk/tests/tutorials/ibor/test_orders.py:25]: your indentation is off by a char!

**Pull request merged**